internal/services/apimanagement/api_management_data_source.go

```diff
@@ -263,6 +291,16 @@ func dataSourceApiManagementRead(d *pluginsdk.ResourceData, meta interface{}) er
 
 	d.Set("sku_name", flattenApiManagementServiceSkuName(resp.Sku))
 
+	if resp.Sku.Name != apimanagement.SkuTypeConsumption {
+		tenantAccessInformationContract, err := tenantAccessClient.ListSecrets(ctx, id.ResourceGroup, id.ServiceName, "access")
+		if err != nil {
+			return fmt.Errorf("retrieving tenant access properties for %s: %+v", *id, err)
+		}
+		if err := d.Set("tenant_access", flattenApiManagementTenantAccessSettings(tenantAccessInformationContract)); err != nil {
+			return fmt.Errorf("setting `tenant_access`: %+v", err)
+		}
+	}
```

Comment:
we need to ensure the `tenant_access` block is always set into the state:

```suggestion
	tenantAccess := make([]interface{}, 0)
	if resp.Sku.Name != apimanagement.SkuTypeConsumption {
		tenantAccessInformationContract, err := tenantAccessClient.ListSecrets(ctx, id.ResourceGroup, id.ServiceName, "access")
		if err != nil {
			return fmt.Errorf("retrieving tenant access properties for %s: %+v", *id, err)
		}
		tenantAccess = flattenApiManagementTenantAccessSettings(tenantAccessInformationContract)
	}
	if err := d.Set("tenant_access", tenantAccess); err != nil {
		return fmt.Errorf("setting `tenant_access`: %+v", err)
	}
```

Reply:
Thanks for you feedback. Fixed.